### ISSUE TITLE
[dart2js] Implement support for generic function instantiation of existing function objects in constants

### ISSUE THREAD

**Issue by eernstg** [dart-lang/sdk#47152]
This is the dart2js specific issue for #47150, which has the details.

**Comment by fishythefish** [comment]
I agree, but I'll leave this open until the CFE implementation lands. Also passing this to @rakudrama, who has been handling other instantiation-related work.

**Comment by devoncarew** [comment]
FYI, the related CFE issue - https://github.com/dart-lang/sdk/issues/47154 - is closed.
